src/openrct2/world/sprite.c

```diff
@@ -285,8 +261,39 @@ static void sprite_reset(rct_unk_sprite *sprite)
 
     sprite->linked_list_type_offset = llto;
     sprite->next = next;
+    sprite->next_in_quadrant = next_in_quadrant;
     sprite->previous = prev;
     sprite->sprite_index = sprite_index;
+    sprite->sprite_identifier = SPRITE_IDENTIFIER_NULL;
+}
+
+/**
+* Clears all the unused sprite memory to zero. Probably so that it can be compressed better when saving.
+*  rct2: 0x0069EBA4
+*/
+void sprite_clear_all_unused()
```

Comment:
The two are very similar, but `sprite_clear_all_unused` is the more conservative version: it explicitly sets `linked_list_type_offset`, `sprite_index` and `next_in_quadrant` (if broken) fields. I don't want to change this behaviour this close to release, are you fine with `sprite_clear_all_unused` calling `reset_empty_sprites`, but then still doing a post-process loop?